### ISSUE TITLE
Dialog creates exceptions when rendered inside shadow DOM with styles loaded

### ISSUE THREAD

**Issue by justin-hackin** [material-components/material-components-web#2212]
### What MDC Web Version are you using?
0.30.0

### What browser(s) is this bug affecting?
Version 63.0.3239.132 (Official Build) Built on Ubuntu , running on Ubuntu 16.04 (64-bit)

UA: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/63.0.3239.132 Chrome/63.0.3239.132 Safari/537.36

### What are the steps to reproduce the bug?
Visit https://jsfiddle.net/o4d8bk24/18/ and run
Notice by changing exposeBug = false, the error does not occur (presumably because styles haven't loaded yet)
### What is the expected behaviour?

Dialog inserted into shadow DOM should open without errors.

### What is the actual behavior?
Error in material-components-web.js
>>7483: nodeComputedStyle = nodeComputedStyle || window.getComputedStyle(node);
Uncaught TypeError: Failed to execute 'getComputedStyle' on 'Window': parameter 1 is not of type 'Element'.
(when node parameter is a document fragment)

Dialog appears as expected (but focus trapping probably breaks based on call stack)

### Any other information you believe would be useful?
I couldn't find your jsbin template before creating this issue.  Consider adding it to https://github.com/material-components/material-components-web/blob/master/CONTRIBUTING.md or the issues header.

**Comment by kfranqueiro** [comment]
Thanks for making us aware of this issue; we'll need to investigate.

One of our engineers took a screenshot of the call stack, for reference:

![image](https://user-images.githubusercontent.com/398379/36049427-bf1acebe-0db0-11e8-9607-5abeb9dfb9f1.png)

FYI, we have a link to a codepen template in our issue template:

> We also have a [template on CodePen](http://codepen.io/traviskaufman/pen/pNQmRp) which we encourage you to use to create a reproduction of
> the issue. The less time it takes for us to repro the issue, the less time it takes to verify and
> fix it!

**Comment by justin-hackin** [comment]
@kfranqueiro yes I saw in your issue template the codepen template but usually people don't open up an issue form until they've already reproduced the error hence my comment.  Thanks for acknowledging this, glad your engineers could reproduce

**Comment by justin-hackin** [comment]
May be related to https://github.com/davidtheclark/focus-trap/issues/39.  As a workaround, I used the dialog's destroy function and it didn't trigger the same bug.  I tried some other solutions like:

**Comment by justin-hackin** [comment]
May be related to https://github.com/davidtheclark/focus-trap/issues/39.  As a workaround, I used the dialog's destroy function and it didn't trigger the same bug.  I tried some other solutions like:

```js
const acceptButton = el.querySelector('.offer-details-dialog__accept-button');
// MDC intereface for creating a custom focus trap does not allow passing additional focus-trap options
// poor design as it requires us to also include focus-trap in project deps and doesn't preserve default options
const focusTrapFactory = dialogEl =>
  focusTrap(dialogEl, {
    initialFocus: acceptButton,
    clickOutsideDeactivates: true,
    returnFocusOnDeactivate: false,
  });
util.createFocusTrapInstance(el, undefined, focusTrapFactory);
```
This did not seem to squelch the error.  It was exceptionally difficult to understand how to disable focus trapping; I spent an afternoon exploring the codebase and still did not succeed.

**Comment by justin-hackin** [comment]
More info on this bug:

close function sets `this.isOpen_` to false, but in the bottom of trace above, inside `this.transitionEndHander_` `thiis.isOpen_` is true.  `this` appears to be a different instance in the former and latter but they appear to be bound to the same context from inspecting the codebase.
